src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
+
+	public void resetBasket() {
+		buyBasket();
+		codeDePromotion = false;
+	}
+
+	public void buyBasket() {
+		map.clear();
+	}
+
+	public boolean isBasketContains(String produit) {
+		boolean found = false;
+		for (String s : map.keySet()) {
+			if (s == produit) found = true;
+		}
+		return found;
+	}
+
+	public void mixWithBasket(BasketInformations b) {
+		map.putAll(b.map);
+	}
```

Comment:
This method does nothing because the 'map' is an static method and is merging the same memory reference, the end result is the map variable ends up not changing. Here we can see the problems mentioned in the commentaries about map being static and the use of hashMap as a data structure to contain more than one of the same items.